### PR TITLE
Fix leaked storage classes in tier 1 storage tests

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
### What this PR does
Before this PR: Some storage tier 1 tests would create storage classes and never remove them
After this PR: No storage classes are leaked from tests

Fixes [CNV-38887](https://issues.redhat.com/browse/CNV-38887)

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place:  [CNV-38887](https://issues.redhat.com/browse/CNV-38887)

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/storage/datavolume.go [hunk 1]
```diff
@@ -1319,7 +1320,10 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 		})
 
 		AfterEach(func() {
-			Expect(virtClient.VirtualMachinePreference(virtualMachinePreference.Namespace).Delete(context.Background(), virtualMachinePreference.Name, metav1.DeleteOptions{})).To(Succeed())
+			Expect(stderror.Join(
```
[the remark this answers is not in the document]
Comment (EduardGomezEscandell) [reply tests/storage/datavolume.go:1323]: I wanted to ensure the second function runs even if the first one fails.

I'm new to gomega so I'm not sure the assertion failure causes the function to exit.

[the remark this answers is not in the document]
Comment (xpivarc) [reply tests/storage/datavolume.go]: It doesn't matter because if the first one fails we are already not in good shape.

**Review by xpivarc — COMMENTED** [review]
/approve

**Review by EdDev — CHANGES_REQUESTED** [review]
I think there are a few issues here:
- The cleanup is not assured if the prev delete fails. It will be better to place the new one in a separate AfterEach. It will be nicer IMO to use `DeferCleanup` after it is created.
- Why this is not part of the general cleanup we do after each test?
- The deletion is async, i.e. when the next test starts, it may yet to be gone. Can this cause a problem or is it ok?
- To keep the history of commits clean, it will be better to squash the two commits. Initial I did not understand what is going on.

File: tests/storage/datavolume.go [hunk 2]
```diff
@@ -1280,7 +1280,7 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 		)
 	})
 
-	Context("With VirtualMachinePreference and PreferredStorageClassName", func() {
+	FContext("With VirtualMachinePreference and PreferredStorageClassName", func() {
```
Comment (EdDev) [inline tests/storage/datavolume.go:1283]: Leftover

Comment (EduardGomezEscandell) [reply tests/storage/datavolume.go]: Woops, good catch

**Review by EdDev — APPROVED** [review]
Thanks!

**Pull request merged**